CA.md

```diff
@@ -0,0 +1,73 @@
+
+Once you have completed the
+[compilation instructions](https://github.com/google/hiba/blob/main/INSTALL.md#compilation)
+you will have a `hiba-gen` and `hiba-ca.sh` in your hiba directory. You can now
+create CA keys, grants, identities, grant eligibility, and sign host and user
+certificates with these extensions. The following assumes you are in your hiba
+directory. The CA structure will be in `~/.hiba-ca` by default.
+
+1. Generate a CA keypair.
+    ```
+    $ ./hiba-ca.sh -c -- -N ""
+    == Generating CA keys in ~/.hiba-ca/g/ca
+    == Done ==
+    ```
+1. Generate a test user keypair (alternatively you can copy existing keys to
+`~/.hiba-ca/users/`).
+    ```
+    $ ./hiba-ca.sh -c -u -I testuser -- -N ""
+    == Generating user keys in ~/.hiba-ca/g/users/testuser
+    == Done ==
+    Identity created: Private: ~/.hiba-ca/users/testuser
+                      Public:  ~/.hiba-ca/users/testuser.pub
+    ```
+    * Alternatively you can copy existing keys with `./hiba-ca.sh -i -u -f
+      path_to_pubkey -I principal`
+1. Copy the OpenSSH server host public key from the OpenSSH server to your
+   machine and import it.
+    ```
+    ./hiba-ca.sh -i -h -f path_to_pubkey -I hostname
+    ```
+1. Create an identity for prod, with a domain of example.com.
+    ```
+    $ ./hiba-gen -i -f "~/.hiba-ca/policy/identities/prod" domain example.com
+    ```
+1. Create a grant called shell with a matching domain.
+    ```
+    $ ./hiba-gen -f "~/.hiba-ca/policy/grants/shell" domain example.com
+    ```
+1. Allow testuser to request this grant (shell).
+    ```
+    $ ./hiba-ca.sh -p -I testuser -H shell
+    == Done ==
+    User testuser is now eligible for [shell]
+    ```
+1. Sign the OpenSSH server public key into a certificate. -I is the hostname you
+   set when importing the host public key.
+   the public key in ~/.hiba-ca/hosts.
+    ```
+    $ ./hiba-ca.sh -s -h -I test -n test -H prod
+    == Signing host key ID test
+    Signed host key ~/.hiba-ca/hosts/test-cert.pub: id "test" serial 1 for test,test valid from 2022-12-14T16:36:00 to 2022-12-14T17:37:42
+    == Done ==
+    Certificate created: ~/.hiba-ca/hosts/test-cert.pub
+    ```
+1. Get a certificate for testuser with a grant extension (shell).
+    ```
+    ./hiba-ca.sh -s -u -I testuser -H shell -- -P ""
+    == Signing user key ID testuser
+    Signed user key ~/.hiba-ca/users/testuser-cert.pub: id "testuser" serial 2 for testuser valid from 2022-12-14T16:37:00 to 2022-12-14T17:38:39
+    == Done ==
+    Certificate created: ~/.hiba-ca/users/testuser-cert.pubrmd.me/iigY1QeGBKr
+    ```
+1. Copy the host certificate to your OpenSSH server which has been configured
+   with `hiba-chk` as the `AuthorizedPrincipalsCommand`. Copy the CA public key
+(~/.hiba-ca/ca.pub) to the OpenSSH server and add it to the TrustedUserCaKeys
+file. See the [configuration
+documentation](https://github.com/google/hiba/blob/main/INSTALL.md#configuration)
+for details. Use the user certificate to log in.
```

Comment:
This should be a new step and you can add the actual `ssh -I` command for new users not knowing how to select an identity for sshing into a target.